app/assets/javascripts/templates/admin/panels/exchange_distributed_products.html.haml

```diff
@@ -11,7 +11,7 @@
 
     .exchange-products
       -# Scope product list based on permissions the current user has to view variants in this exchange
-      .exchange-product{'ng-repeat' => 'product in supplied_products | filter:productSuppliedToOrderCycle | visibleProducts:exchange:order_cycle.visible_variants_for_outgoing_exchanges | orderBy:"name"' }
+      .exchange-product{'ng-repeat' => 'product in enterprises[exchange.enterprise_id].supplied_products | filter:visibleProducts:exchange:order_cycle.visible_variants_for_outgoing_exchanges | orderBy:"name"' }
```

Comment:
this line change is the single point of this PR.
- supplied_products used to contain all products from all enterprises, it's not available any longer.
- now we have enterprises[exchange.enterprise_id].supplied_products that contains the supplied products for each enterprise and is loaded from the server only when the user opens the products panel and is already filtered on the server (no need for expensive productSuppliedToOrderCycle)

Reply:
Awesome! Is there any chance we can strip out some of that Angular filtering on `visibleProducts` at some point as well?

Reply:
I have a feeling it's something we want to do server-side instead of in the browser...

Reply:
yes!
I left these for later to avoid huge PRs. I created issues now:
visible (filter only) - https://github.com/openfoodfoundation/openfoodnetwork/issues/4480
editable (these need a new flag on each variant) - https://github.com/openfoodfoundation/openfoodnetwork/issues/4481